drivers/spi/spi_shell.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2024 Astroligt
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <string.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/sys/util.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_shell, CONFIG_SPI_LOG_LEVEL);
+
+#define ARGV_DEV	(1)
+#define ARGV_BYTES	(2)
+
+/* Maximum bytes we can write and read at once */
+#define MAX_SPI_BYTES	MIN((CONFIG_SHELL_ARGC_MAX - ARGV_BYTES), 32)
+
+static uint32_t frequency = 1000000;
+
+static void device_name_get(size_t idx, struct shell_static_entry *entry)
+{
+	const struct device *dev = shell_device_lookup(idx, "spi");
+
+	entry->syntax = (dev != NULL) ? dev->name : NULL;
+	entry->handler = NULL;
+	entry->help = NULL;
+	entry->subcmd = NULL;
+}
+
+SHELL_DYNAMIC_CMD_CREATE(dsub_device_name, device_name_get);
+
+static int cmd_spi_transceive(const struct shell *shell_ctx,
+			     size_t argc, char **argv)
+{
+	const struct device *dev;
+	uint8_t rx_buffer[MAX_SPI_BYTES] = {0};
+	uint8_t tx_buffer[MAX_SPI_BYTES] = {0};
+
+	dev = device_get_binding(argv[ARGV_DEV]);
+
+	if (!dev) {
+		shell_error(shell_ctx, "SPI: Device %s not found.",
+			    argv[ARGV_DEV]);
+		return -ENODEV;
+	}
+
+	int bytes_to_send = argc - ARGV_BYTES;
+
+	for (int i = 0; i < bytes_to_send; i++) {
+		tx_buffer[i] = strtol(argv[ARGV_BYTES + i], NULL, 16);
+	}
+
+	const struct spi_buf tx_buffers = {.buf = tx_buffer, .len = bytes_to_send};
+	const struct spi_buf rx_buffers = {.buf = rx_buffer, .len = bytes_to_send};
+
+	const struct spi_buf_set tx_buf_set = {.buffers = &tx_buffers, .count = 1};
+	const struct spi_buf_set rx_buf_set = {.buffers = &rx_buffers, .count = 1};
+
+	const struct spi_config config = {.frequency = frequency,
+					  .operation = SPI_OP_MODE_MASTER | SPI_WORD_SET(8)};
+
+	int ret = spi_transceive(dev, &config, &tx_buf_set, &rx_buf_set);
+
+	if (ret < 0) {
+		shell_error(shell_ctx, "spi_transceive returned %d", ret);
+	} else {
+		ret = 0;
```

Comment:
it is supporting master mode only here so ret will be either 0 or < 0, so you can remove that else block.